### DISCUSSION TITLE
iOS not receiving push notifications (Android working fine) - How to debug?

### DISCUSSION THREAD

**Discussion by rubenferreira97** [firebase/flutterfire#7150] [Q&A]
Hi, I am having problems on receiving notifications on iOS either sent by firebase console (notification test by specifying an apnsToken) or by a curl request using the legacy HTTP Protocol (only notification, no iOS or Android payload sent).
On Android both methods work fine and the notifications are received.

I have followed all the specified steps on the documentation without success:
- Enable push notifications and background modes (background fetch and remote notifications)
- Register a key and export .r8 to firebase console APN Auth keys.
- Register a APP ID (with push notifications enabled).
- Generated a provision profile.
- Await for Firebase initialisation on flutter side.
- Request for notifications permissions on iOS

More info:
I can get APNs IDs on both iOS and Android, so I guess the problem is not in firebase (core) configuration but maybe on firebase messaging. However, the iOS APN (only caps lock characters and numbers) seems really different from a Android APN. Is it normal?

I did not mess with FirebaseAppDelegateProxyEnabled, even thought I receive the following message: FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers.

I know that the curl request returns 200 OK (response code and I am receiving the notification on Android).

Any step that I am forgetting? How can I debug this?

**Comment by y-alkhateeb** [comment]
+1 the same issue
